mcp/shared.go

```diff
@@ -194,7 +195,10 @@ type methodInfo struct {
 // - R: results
 
 // A typedMethodHandler is like a MethodHandler, but with type information.
-type typedMethodHandler[S Session, P Params, R Result] func(context.Context, S, P) (R, error)
+type (
+	typedClientMethodHandler[P Params, R Result] func(context.Context, *ClientRequest[P]) (R, error)
```

Comment:
These seem like they could be one type with a third type parameter.

Reply:
I think there is a lot of cleanup of unexported code I could do. Tomorrow I'll follow up on this and others.